pkg/operator/controller.go

```diff
@@ -148,11 +147,50 @@ func (c *Controller) eventProcessor() {
 
 func (c *Controller) sync(key wqKey) error {
 	var (
-		cr  *tunedv1.Tuned
-		err error
+		cr           *tunedv1.Tuned
+		err, lastErr error
 	)
 	klog.V(2).Infof("sync(): Kind %s: %s/%s", key.kind, key.namespace, key.name)
 
+	if key.kind == wqKindTuned && key.name == tunedv1.TunedDefaultResourceName {
+		// default Tuned changed or a bootstrap event received
+		cr, err = c.syncTunedDefault()
+		if err != nil {
+			return fmt.Errorf("failed to sync default Tuned CR: %v", err)
+		}
+	} else {
+		cr, err = c.listers.TunedResources.Get(tunedv1.TunedDefaultResourceName)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				cr, err = c.syncTunedDefault()
+				if err != nil {
+					return fmt.Errorf("failed to sync default Tuned CR: %v", err)
+				}
+			} else {
+				return fmt.Errorf("failed to get Tuned %s: %v", tunedv1.TunedDefaultResourceName, err)
+			}
+		}
+	}
+	// We have the default Tuned custom resource (cr)
+
+	switch cr.Spec.ManagementState {
+	case operatorv1.Force:
+		// Use the same logic as Managed.
+	case operatorv1.Managed, "":
+		// Managed means that the operator is actively managing its resources and trying to keep the component active.
+	case operatorv1.Removed:
+		// Removed means that the operator is actively managing its resources and trying to remove all traces of the component.
+		lastErr = c.removeResources()
+		goto out
```

Comment:
Haven't seen a lot of `goto`, interesting use.

Reply:
I thought you'd mention this :) http://wiki.c2.com/?StructuredProgrammingWithGoToStatements
I believe that `goto`s to quit a function in this manner are quite frequent.  At least in C programs.  If you can think of a better way to eliminate code duplication in this case, I'd be happy to hear!

Reply:
Nope, not looking to eliminate :smile: